Formula/go.rb

```diff
@@ -79,16 +79,6 @@ def install
     end
   end
 
-  def caveats; <<-EOS.undent
```

Comment:
@ilovezfs I think we can remove the "As of go 1.2" at this point. Personally I think we should just remove all the caveats now as they aren't Homebrew specific.